### PR TITLE
First design

### PR DESCRIPTION
**Pull request by abygeorge7**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1,31 +1,91 @@
...
+    <!-- CDN for Vue.js -->
+    <script src="https://cdn.jsdelivr.net/npm/vue@2/dist/vue.js"></script>
+
+    <!-- CDN for Bulma -->
+    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bulma@0.9.3/css/bulma.min.css" />
+</head>
+
+<body>
+    <div id="app">
+        <section class="hero is-primary trial">
```
Comment (BenjaminSao) [inline index.html]: What is this "trial" class?

File: index.html [hunk 2]
```diff
@@ -1,31 +1,91 @@
...
+            <div class="hero-title">
+                <h1 class=" title is-4 pt-4 pb-0 has-text-centered-desktop">{{title}}</h1>
+            </div>
+            <div class="hero-body pt-0 pb-2 is-flex is-justify-content-space-between is-align-items-center">
+                <div >
+                   <h1 class="is-size-5 pt-0">Net {{netWorth}} : {{incomeTotal}} {{currency}}</h1>
```
Comment (BenjaminSao) [inline index.html]: The "currency" should come before the value.

File: index.html [hunk 3]
```diff
@@ -1,31 +1,91 @@
...
+                    <h1 class="is-size-5 pt-0">Net {{Expense}} : {{expenseTotal}} {{currency}} </h1>
+                    <h1 class="is-size-5 pt-0">Net in hand : {{netInHand}} {{currency}}</h1>
+                </div>
+                <div class="is-flex is-justify-content-flex-end ">
+                    <h1 class="is-size-6 pt-1 pr-1">Currency</h1>
+                    <form action="" >
```
Comment (BenjaminSao) [inline index.html]: This form has no action. Get rid of `action=""`

File: index.html [hunk 4]
```diff
@@ -1,31 +1,91 @@
...
+                        <select class="select is-rounded" v-model= "currencyType" @change="changeCurrency">
```
Comment (BenjaminSao) [inline index.html]: I don't this is-rounded is being applied here.

File: index.html [hunk 5]
```diff
@@ -1,31 +1,91 @@
...
+                            <option>INR</option>
+                            <option>THB</option>
+                        </select>
+                    </form>
+                </div>
+            </div>
+        </section>
+        <section class="container">
+            <h2 class="title is-5 pt-4">Enter transactions below</h2>
+            <Transactions @transaction-submitted="addTransaction" @update-income="calculateIncome"
```
Comment (BenjaminSao) [inline index.html]: HTML is not cap sensitive, so in general we do not capitalize component names. `Transactions` should be changed to `transactions`

Comment (BenjaminSao) [reply index.html]: This `Transactions` component has the wrong name for it's purpose. This is for the adding for transactions not the list of transactions. Can you think of a better name?

File: index.html [hunk 6]
```diff
@@ -1,31 +1,91 @@
...
+                @update-expense="calculateExpense" @transaction-edited="clearPreviousId" @update-currency ="changeCurrency"
```
Comment (BenjaminSao) [inline index.html]: `@transaction-edited` this event calling `clearPreviousId` is kinda off. You should not base this event on a specific procedure in the view. `@transaction-edited` should invoke a general method rather than a specific one like `clearPreviousId`.

File: index.html [hunk 7]
```diff
@@ -1,31 +1,91 @@
...
+                :final-transaction="finalTransaction" :edit-transaction-id="editTransactionId"  
+                ></Transactions>
+            <h2 class="title is-6 pt-2">Transactions Summary</h2>
```
Comment (BenjaminSao) [inline index.html]: Keep the styles, consistent. When designing UI, you do not want to many variation, it'll look messy. The above `Transactions` component uses `is-5` and this one uses `is-6` is there really a reason to switch between the two?

File: index.html [hunk 8]
```diff
@@ -1,31 +1,91 @@
...
+            <div class="columns">
+                <div class="column">
+                    <div class="card ">
+                        <footer class="card-footer">
+                            <p class="card-footer-item">Transactions</p>
+                            <p class="card-footer-item">Description</p>
+                            <p class="card-footer-item">Credit/Debit</p>
+                            <p class="card-footer-item">Amount $</p>
```
Comment (BenjaminSao) [inline index.html]: Currency should be at the start.

File: index.html [hunk 9]
```diff
@@ -1,31 +1,91 @@
...
+                            <p class="card-footer-item is-small">Delete</p>
+                            <p class="card-footer-item is-small">Edit</p>
+                        </footer>
+                        <footer class="card-footer " v-for="(item, index) in finalTransaction">
+                            <p class="card-footer-item"> {{item.name}}</p>
+                            <p class="card-footer-item">{{item.description}}</p>
+                            <p class="card-footer-item">{{item.type}}</p>
+                            <p class="card-footer-item"> {{item.amount}} $</p>
+                            <p class="card-footer-item" @click="deleteTransaction(index);
```
Comment (BenjaminSao) [inline index.html]: This'll work for now. But in the future you may want to delete by `id`.

Comment (BenjaminSao) [reply index.html]: Consider the case, you want to delete it from a DB or something, you can't reference it by the index listed here, you'll have to use the id.

File: index.html [hunk 10]
```diff
@@ -1,31 +1,91 @@
...
+                      calculateExpense();calculateIncome();changeCurrency();"><span><button class="delete"></span></button>
+                            <p>
```
Comment (BenjaminSao) [inline index.html]: What's up with this `p` tag?

File: index.html [hunk 11]
```diff
@@ -1,31 +1,91 @@
...
+                    <div class="card ">
+                        <footer class="card-footer">
+                            <p class="card-footer-item">Transactions</p>
+                            <p class="card-footer-item">Description</p>
+                            <p class="card-footer-item">Credit/Debit</p>
+                            <p class="card-footer-item">Amount $</p>
+                            <p class="card-footer-item is-small">Delete</p>
+                            <p class="card-footer-item is-small">Edit</p>
+                        </footer>
+                        <footer class="card-footer " v-for="(item, index) in finalTransaction">
```
Comment (BenjaminSao) [inline index.html]: instead of `item`, change to `transaction`

Comment (BenjaminSao) [reply index.html]: `finalTransaction` is a bad name. Maybe `transactions` plural would be better.

File: index.html [hunk 12]
```diff
@@ -1,31 +1,91 @@
...
+                            <p class="card-footer-item"> {{item.name}}</p>
+                            <p class="card-footer-item">{{item.description}}</p>
+                            <p class="card-footer-item">{{item.type}}</p>
+                            <p class="card-footer-item"> {{item.amount}} $</p>
+                            <p class="card-footer-item" @click="deleteTransaction(index);
+                      calculateExpense();calculateIncome();changeCurrency();"><span><button class="delete"></span></button>
```
Comment (BenjaminSao) [inline index.html]: The view has too much logic. The view shouldn't have too much knowledge of what logical stuff it should do. Make this call one method inside the view-model. Read about the MVVM model.

File: index.html [hunk 13]
```diff
@@ -1,31 +1,91 @@
...
-
-        <script src="./main.js"></script>
-    </body>
-</html>
+            </div>
+        </section>
+    </div>
+    <script type="text/x-template" id="transactions-template">
+        <div>
+        <form action="" @submit.prevent="onSubmit"  class ="field has-addons">
```
Comment (BenjaminSao) [inline index.html]: Useless `action=""`.

File: index.html [hunk 14]
```diff
@@ -1,31 +1,91 @@
...
+            <input type="text" class="input is-primary" v-model= "name" placeholder = " Enter Name of transaction"  required>
```
Comment (BenjaminSao) [inline index.html]: Can you make sure your coding style is better. `placeholder = " Enter Name of transaction"` -> `placeholder="Enter Name of transaction"`

Comment (BenjaminSao) [reply index.html]: Work on spacing...

File: index.html [hunk 15]
```diff
@@ -1,31 +1,91 @@
...
+            <input type="text" class="input is-primary" id="description" v-model= "description"placeholder = " Enter Details" required >
```
Comment (BenjaminSao) [inline index.html]: Here's another instance. Refactoring & make code look cleaner.

File: main.js [hunk 16]
```diff
@@ -1,11 +1,188 @@
-var app = new Vue({
-    el: "#app",
-    data: {
-        nums: []
+
+
+
+Vue.component('Transactions', {
```
Comment (BenjaminSao) [inline main.js]: Component's name should be lowercase. HTML is not cap sensitive.

File: main.js [hunk 17]
```diff
@@ -1,11 +1,188 @@
...
+    template: `#transactions-template`,
+    props:
+    {
+        finalTransaction: Array,
```
Comment (BenjaminSao) [inline main.js]: bad naming. Should be `transactions`

File: Images/edit-3.svg [hunk 18]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="18" height="18" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-edit-3"><path d="M12 20h9"></path><path d="M16.5 3.5a2.121 2.121 0 0 1 3 3L7 19l-4 1 1-4L16.5 3.5z"></path></svg>
```
Comment (BenjaminSao) [inline Images/edit-3.svg:1]: Change `Images` folder to `images`

File: index.html [hunk 19]
```diff
@@ -1,31 +1,91 @@
...
+                        </footer>
+                        <footer class="card-footer " v-for="(item, index) in finalTransaction">
+                            <p class="card-footer-item"> {{item.name}}</p>
+                            <p class="card-footer-item">{{item.description}}</p>
+                            <p class="card-footer-item">{{item.type}}</p>
+                            <p class="card-footer-item"> {{item.amount}} $</p>
+                            <p class="card-footer-item" @click="deleteTransaction(index);
+                      calculateExpense();calculateIncome();changeCurrency();"><span><button class="delete"></span></button>
+                            <p>
+                            <p class="card-footer-item is-small" @click="editTransaction(item)"> <img src="./Images/edit-3.svg" alt=""></p>
```
Comment (BenjaminSao) [inline index.html]: `Images` -> `images`

File: main.js [hunk 20]
```diff
@@ -1,11 +1,188 @@
...
+        editTransactionId: function (_editTransactionId)
+        {
+            if (_editTransactionId)
+            {
+                const transaction = this.finalTransaction.find(t => t.id === _editTransactionId);
+                this.name = transaction.name;
+                this.description = transaction.description;
+                this.type = transaction.type;
+                this.amount = transaction.amount;
+                return;
```
Comment (BenjaminSao) [inline main.js]: Useless return

File: main.js [hunk 21]
```diff
@@ -1,11 +1,188 @@
...
+            }
         }
     },
-});
\ No newline at end of file
+    methods: {
+
+        editTransactionDisplay (){
+            this.$emit('console.log');
+        } ,
```
Comment (BenjaminSao) [inline main.js]: What is this?

File: main.js [hunk 22]
```diff
@@ -1,11 +1,188 @@
...
+
+            if (this.editTransactionId) {
+                transaction = this.finalTransaction.find(t => t.id === this.editTransactionId);
+                console.log(transaction.id, transaction.name);
+                transaction.name = this.name;
+                transaction.description = this.description;
+                transaction.type = this.type;
+                transaction.amount = this.amount;
+                this.$emit("transaction-edited");
+                this.$emit('update-income');
```
Comment (BenjaminSao) [inline main.js]: double quotes

File: main.js [hunk 23]
```diff
@@ -1,11 +1,188 @@
...
+                this.$emit('update-expense');
+                this.$emit('update-currency');
+                this.name = null;
+                this.description = null;
+                this.type = null;
+                this.amount = null;
+            }
+            else {
+                let transactionList = {
```
Comment (BenjaminSao) [inline main.js]: This is not a list. This is a new `transaction`.

File: main.js [hunk 24]
```diff
@@ -1,11 +1,188 @@
...
+                }
+                this.$emit('transaction-submitted', transactionList);
+                this.$emit('update-income');
+                this.$emit('update-expense');
+                this.$emit('update-currency');
+                this.name = null;
+                this.description = null;
+                this.type = null;
+                this.amount = null;
+                console.log(this.finalTransaction);
```
Comment (BenjaminSao) [inline main.js]: Get rid of all console.log... If this was a production application, this shouldn't be here.

File: main.js [hunk 25]
```diff
@@ -1,11 +1,188 @@
...
+    
+
+
+var app = new Vue
+    ({
+        el: "#app",
+        data: {
+            title: "Expense Tracker",
+            netWorth: "Income",
+            Expense: "Expense",
```
Comment (BenjaminSao) [inline main.js]: Why are these here?

Comment (BenjaminSao) [reply main.js]: Can't you just leave them in the `view?

File: main.js [hunk 26]
```diff
@@ -1,11 +1,188 @@
...
+            
+            addTransaction(transactionList) {
+
+                this.finalTransaction.push(transactionList);
+            },
+
+            calculateIncome() {
+                
+                let temp = 0;
+                console.log("called for value update");
```
Comment (BenjaminSao) [inline main.js]: no `console.log`

File: main.js [hunk 27]
```diff
@@ -1,11 +1,188 @@
...
+                    for (let item in this.finalTransaction) {
```
Comment (BenjaminSao) [inline main.js]: `let transaction in this.transactions`

File: main.js [hunk 28]
```diff
@@ -1,11 +1,188 @@
...
+                            temp += this.finalTransaction[item].amount;
+                    }
+                
+                this.tempIncomeTotal = temp;
+
+            },
+            calculateExpense() {
+                
+                let temp = 0;
+                    console.log("called for value update")
```
Comment (BenjaminSao) [inline main.js]: again `console.log`

File: main.js [hunk 29]
```diff
@@ -1,11 +1,188 @@
...
+            },
+            
+            clearPreviousId()
+            {
+                this.editTransactionId = null;
+            },
+            editTransactionDisplay()
+            {
+                console.log("to display edit");
+            }
```
Comment (BenjaminSao) [inline main.js]: Unnecessary

**Review by BenjaminSao — CHANGES_REQUESTED** [review]

File: index.html (hunk 1 above)
Comment (abygeorge7) [reply index.html]: deleted

File: index.html (hunk 2 above)
Comment (abygeorge7) [reply index.html]: ok

File: index.html (hunk 3 above)
Comment (abygeorge7) [reply index.html]: yes done

File: index.html (hunk 4 above)
Comment (abygeorge7) [reply index.html]: removed

File: index.html (hunk 5 above)
Comment (abygeorge7) [reply index.html]: ok renamed to transaction. Its a debit or credit and I think transaction is suitable

Comment (abygeorge7) [reply index.html]: or please put any other suggestion

File: index.html (hunk 6 above)
Comment (abygeorge7) [reply index.html]: Ok i renamed it to UpdateID. Hope that was what you mentioned

File: index.html (hunk 7 above)
Comment (abygeorge7) [reply index.html]: Ok lets keep is-6 .updated that style

File: index.html (hunk 8 above)
Comment (abygeorge7) [reply index.html]: yes. updated

File: index.html (hunk 10 above)
Comment (abygeorge7) [reply index.html]: you mean i have to use another tag .not p?

File: index.html (hunk 11 above)
Comment (abygeorge7) [reply index.html]: first one is resolved and second one that is changing to finaltransaction is making it less distinguishable  as both transaction and transactions are looking similar and making it difficult to read and understand code.Do we still need to change it?

File: index.html (hunk 13 above)
Comment (abygeorge7) [reply index.html]: done

File: index.html (hunk 14 above)
Comment (abygeorge7) [reply index.html]: done

File: index.html (hunk 15 above)
Comment (abygeorge7) [reply index.html]: done

File: main.js (hunk 16 above)
Comment (abygeorge7) [reply main.js]: ok done

File: main.js (hunk 17 above)
Comment (abygeorge7) [reply main.js]: ok

File: Images/edit-3.svg (hunk 18 above)
Comment (abygeorge7) [reply Images/edit-3.svg:1]: done

File: index.html (hunk 19 above)
Comment (abygeorge7) [reply index.html]: yes done

File: main.js (hunk 20 above)
Comment (abygeorge7) [reply main.js]: done

File: main.js (hunk 21 above)
Comment (abygeorge7) [reply main.js]: was experimenting.deleted

File: main.js (hunk 22 above)
Comment (abygeorge7) [reply main.js]: done

File: main.js (hunk 23 above)
Comment (abygeorge7) [reply main.js]: done

File: main.js (hunk 24 above)
Comment (abygeorge7) [reply main.js]: done

File: main.js (hunk 25 above)
Comment (abygeorge7) [reply main.js]: ok done

File: main.js (hunk 26 above)
Comment (abygeorge7) [reply main.js]: done

File: main.js (hunk 27 above)
Comment (abygeorge7) [reply main.js]: ok

File: main.js (hunk 28 above)
Comment (abygeorge7) [reply main.js]: deleted

File: main.js (hunk 29 above)
Comment (abygeorge7) [reply main.js]: deleted

File: index.html (hunk 10 above)
Comment (BenjaminSao) [reply index.html]: No, it's not used. Is it supposed to be `</p>`

File: main.js (hunk 17 above)
Comment (BenjaminSao) [reply main.js]: change `finalTransaction` to `transactions`

File: main.js (hunk 22 above)
Comment (BenjaminSao) [reply main.js]: I don't see the changes. Change `'update-income'` to `"update-income"`

File: main.js (hunk 27 above)
Comment (BenjaminSao) [reply main.js]: There seems to be no changes.

File: main.js [hunk 30]
```diff
@@ -1,11 +1,205 @@
...
+                    return "₹"
+                if (this.currencyType === 'THB')
+                    return "฿" 
+            },
+            
+                
+        },
+           
+        
+    });
```
Comment (BenjaminSao) [inline main.js]: What's up with the spacing?

File: index.html (hunk 9 above)
Comment (abygeorge7) [reply index.html]: was updated

File: index.html (hunk 12 above)
Comment (abygeorge7) [reply index.html]: i did some changes .Please review

**Review by BenjaminSao — APPROVED** [review]
